sdk/core/azure-core/src/main/java/com/azure/core/http/policy/OverrideHeaderPolicy.java

```diff
@@ -0,0 +1,39 @@
+package com.azure.core.http.policy;
+
+import com.azure.core.http.HttpHeader;
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.util.Context;
+import com.azure.core.http.HttpRequest;
+import static com.azure.core.util.Context.AZURE_REQUEST_HTTP_HEADERS_KEY;
+import reactor.core.publisher.Mono;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * The pipeline policy that override or add  {@link HttpHeader} in {@link HttpRequest} by reading values from
+ * {@link Context} with key  'azure-request-override-http-headers-key'. The value for this key should be of type
+ * {@link HttpHeaders} for it to be added in {@link HttpRequest}.
+ */
+public class OverrideHeaderPolicy implements HttpPipelinePolicy {
+
+    @Override
+    public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+
+        Optional<Object> customHttpHeadersObject = context.getData(AZURE_REQUEST_HTTP_HEADERS_KEY);
+        if (!Objects.isNull(customHttpHeadersObject) && customHttpHeadersObject.get() instanceof HttpHeaders) {
```

Comment:
Context will return an `empty` Optional if key does not exists, so we can check for the presence and if so check for the type, I think no need to check for nullability as `null instanceof HttpHeaders ` evaluated to false

```java
if (customHttpHeadersObject.isPresent() 
    && customHttpHeadersObject.get() instanceof HttpHeaders) {
```